### ISSUE TITLE
additions to jaq/jaq-std/src/std.jq from jq's builtin.jq

### ISSUE THREAD

**Issue by pkoppstein** [01mf02/jaq#33]
Some functions in jq's builtin.jq work perfectly well in jaq and so according to my understanding could easily be made available to jaq users by copying them into jaq's std.jq

Some examples I've tested:
```
def INDEX(stream; idx_expr):
  reduce stream as $row ({}; .[$row|idx_expr|tostring] = $row);

def normals: select(isnormal);

def finites: select(isfinite);

def IN(s): any(s == .; .);

def IN(src; s): any(src == s; .);
```

Also, bsearch/1 can easily be adapted by changing `$target` to `target`.

**Comment by 01mf02** [comment]
Okay, I can see that `INDEX` works, but I have no intent of adding SQL-style operators, as written in the README.

Regarding `combinations`, how did you make these compile in jaq? `combinations` is defined recursively, so that yields the following error for me:

~~~
Error: could not find function
   ╭─[<unknown>:4:22]
   │
 4 │           | (.[1:] | combinations) as $y
   ·                      ──────┬─────  
   ·                            ╰─────── could not find function
───╯
~~~

**Comment by pkoppstein** [comment]
@01mf02 - My apologies for erroneously including `combinations`.  I've added some others that I have tested.

**Comment by 01mf02** [comment]
Ok, I have now included `normals` and `finites` in 0537323d5837cec64e5b407115db7d9b3de600d2. Thanks for your suggestion!

Thanks also for finding that `bsearch` compiles correctly; however, because the parser takes too long for my taste to parse such large expressions (~56 seconds on my machine for `bsearch`, see also #7), I would like to postpone integrating such filters until the parser is faster.

The other filters (`IN`, `INDEX`) are SQL-style operators which, as I have already mentioned, I do not wish to include in the standard library (because I do not like some filters to be uppercase, others lowercase etc.).
However, I could imagine something like an `sql.jq` module for SQL-style operators; but then, this would not be usable right now because jaq does not support modules.

**Comment by 01mf02** [comment]
Just for reference, I have just sped up the parser in 012c9c5, so filters like `bsearch` should be feasible to implement now.
However, I am more inclined to keep the standard library small, so I do not like to include a filter like `bsearch` in it.
